examples/filters/colorfilter/color_filter.go

```diff
@@ -0,0 +1,152 @@
+// Package colorfilter implements a modular camera that filters the output of an underlying camera and only keeps
+// captured data if the vision service detects a certain color in the captured image.
+package colorfilter
+
+import (
+	"context"
+	"fmt"
+	"image"
+	"time"
+
+	"github.com/edaniels/golog"
+	"github.com/pkg/errors"
+	"github.com/viamrobotics/gostream"
+
+	"go.viam.com/rdk/components/camera"
+	"go.viam.com/rdk/data"
+	"go.viam.com/rdk/pointcloud"
+	"go.viam.com/rdk/resource"
+	"go.viam.com/rdk/rimage/transform"
+	"go.viam.com/rdk/services/vision"
+)
+
+var (
+	Model            = resource.NewModel("example", "filter", "colorfilter")
+	errUnimplemented = errors.New("unimplemented")
+)
+
+func init() {
+	resource.RegisterComponent(camera.API, Model, resource.Registration[camera.Camera, *Config]{
+		Constructor: newCamera,
+	})
+}
+
+func newCamera(ctx context.Context, deps resource.Dependencies, conf resource.Config, logger golog.Logger) (camera.Camera, error) {
+	c := &colorFilterCam{
+		Named:  conf.ResourceName().AsNamed(),
+		logger: logger,
+	}
+	if err := c.Reconfigure(ctx, deps, conf); err != nil {
+		return nil, err
+	}
+	return c, nil
+}
+
+type Config struct {
+	ActualCam     string `json:"actual_cam"`
+	VisionService string `json:"vision_service"`
+}
+
+func (cfg *Config) Validate(path string) ([]string, error) {
+	if cfg.ActualCam == "" {
+		return nil, fmt.Errorf(`expected "actual_cam" attribute in %q`, path)
+	}
+
+	return []string{cfg.ActualCam}, nil
+}
+
+type colorFilterCam struct {
+	resource.Named
+	actualCam     camera.Camera
+	visionService vision.Service
+	logger        golog.Logger
+}
+
+// resource.Resource methods
+func (c *colorFilterCam) Reconfigure(ctx context.Context, deps resource.Dependencies, conf resource.Config) error {
```

Comment:
Also pretty nitpicky, but the Go standard is to have the comments above public methods start with the name of the method and describe what they do. Since this is an example that customers will be following, we should probably follow that. A lot of this code is boilerplate to implement various interfaces, so even just something like `Reconfigure implements the resource.Resource interface, which is needed to satisfy the camera.Camera interface` or something like that